cocotb/share/lib/vpi/VpiTrampoline.cpp

```diff
@@ -0,0 +1,228 @@
+        static PLI_INT32  vpi_flush           (void)                                                                               { static fptr_vpi_flush                 f = reinterpret_cast<fptr_vpi_flush>                (resolve_function("vpi_flush"                )); return f();                                   }
+        static PLI_INT32  vpi_mcd_flush       (PLI_UINT32 mcd)                                                                     { static fptr_vpi_mcd_flush             f = reinterpret_cast<fptr_vpi_mcd_flush>            (resolve_function("vpi_mcd_flush"            )); return f(mcd);                                }
+        static PLI_INT32  vpi_control         (PLI_INT32 operation, ...)                                                           { va_list ap; va_start(ap, operation); PLI_INT32 ret = vpi_vcontrol(operation, ap); va_end(ap); return ret; }
+        static vpiHandle  vpi_handle_by_multi_index (vpiHandle obj, PLI_INT32 num_index, PLI_INT32 *index_array)                   { static fptr_vpi_handle_by_multi_index f = reinterpret_cast<fptr_vpi_handle_by_multi_index>(resolve_function("vpi_handle_by_multi_index")); return f(obj, num_index, index_array);        }
+
+        static PLI_INT32 vpi_vcontrol         (PLI_INT32 operation, va_list ap)
+        {
+            static fptr_vpi_control f = reinterpret_cast<fptr_vpi_control>(resolve_function("vpi_control"));
+
+            switch(operation)
+            {
+                case vpiStop:
+                    return f(operation, va_arg(ap, PLI_INT32));
+                case vpiFinish:
+                    return f(operation, va_arg(ap, PLI_INT32));
+                case vpiReset:
+                    return f(operation, va_arg(ap, PLI_INT32), va_arg(ap, PLI_INT32), va_arg(ap, PLI_INT32));
+                case vpiSetInteractiveScope:
+                    return f(operation, va_arg(ap, vpiHandle));
+                default:
+                    LOG_CRITICAL("Unable to forward, operation %" PRId32 " unknown", operation);
+            }
+        }
+
+    private:
+        VpiTrampoline() {}
+
+        VpiTrampoline(VpiTrampoline const&);
+        void operator=(VpiTrampoline const&);
+
+        static void *resolve_function(const char *name)
+        {
+#if defined(ALDEC)
+            static HMODULE hModule = GetModuleHandle("aldecpli.dll");
+#elif defined(GHDL)
+            static HMODULE hModule = GetModuleHandle("libghdlvpi.dll");
+#elif defined(ICARUS)
+            static HMODULE hModule = GetModuleHandle("vvp.exe");
+#elif defined(MODELSIM)
+            static HMODULE hModule = GetModuleHandle("mtipli.dll");
+#else
+#error No target module defined for trampoline
+#endif
```

Comment:
Is this any better than having the names in the `.def` files?

Reply:
This is here for compatibility to not break any of the existing tools. The symbols are resolved here in the same way as the OS does. In the future the module names can be removed by searching for `vpi_*` in the whole process space but this works only if there is only one module exposing `vpi_*` as otherwise it would be ambiguous which to select. As a consequence this requires testing in all tools and is therefore left for a future PR.

Reply:
Can you add this piece of information to the function as comment please?